docs/developer_guide.md

```diff
@@ -0,0 +1,75 @@
+# Developer guide
+
+## Leapp tool
```

Comment:
How about adding more info about what the plugin does, e.g:

The plugin enables the consumer to use a job template to upgrade Red Hat Enterprise Linux 7 hosts to Red Hat Enterprise Linux 8.

example usage:
- In the Foreman web UI, navigate to Hosts > All Hosts.
- Select the hosts that you want to upgrade to Red Hat Enterprise Linux 8.
- In the upper right of the Hosts window, from the Select Action list, select **Preupgrade check with Leapp**.
- Click Submit to start the pre-upgrade check.
- When the check is finished, click the **Leapp preupgrade report tab** to continue

Read more about it [here](https://access.redhat.com/documentation/en-us/red_hat_satellite/6.10/html/managing_hosts/upgrading-hosts-from-rhel7-to-rhel8_managing-hosts)